benchmark/src/jvmMain/kotlin/benchmark/Benchmark.kt

```diff
@@ -0,0 +1,164 @@
+    var pooleValid: Boolean = false
+}
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(BenchmarkTimeUnit.MILLISECONDS)
+open class Benchmark {
+    lateinit var state: List<Pair<InputRecord, ResultRecord>>
+
+    @Setup
+    fun prepare() {
+        state =
+            SystemFileSystem.source(Path("../sample-data/opening_hours_counts.tsv")).buffered().use {
+                source ->
+                Tsv.decodeFromSource<InputRecord>(source).map { it to ResultRecord() }.toList()
+            }
+    }
+
+    @Benchmark
+    fun parse() = state.forEach { (input, result) ->
+        input.times.toOpeningHoursOrNull(lenient = true)?.let { oh -> result.parsed = oh }
+    }
+
+    @Benchmark
+    fun parseStrict() = state.forEach { (input, result) ->
+        input.times.toOpeningHoursOrNull(lenient = true)?.let { _ -> result.valid = true }
+    }
+
+    @Benchmark
+    fun create() = state.forEach { (_, result) -> result.created = result.parsed?.toString() }
+
+    @Benchmark
+    fun pooleParse() = state.forEach { (input, result) ->
+        result.pooleParsed =
+            try {
+                ch.poole.openinghoursparser
+                    .OpeningHoursParser(input.times.byteInputStream())
+                    .rules(false)
+            } catch (e: ch.poole.openinghoursparser.ParseException) {
+                null
+            }
+    }
+
+    @Benchmark
+    fun pooleParseStrict() = state.forEach { (input, result) ->
+        val rules =
+            try {
+                ch.poole.openinghoursparser
+                    .OpeningHoursParser(input.times.byteInputStream())
+                    .rules(true)
+            } catch (e: ch.poole.openinghoursparser.ParseException) {
+                null
+            }
+        if (rules != null) result.pooleValid = true
+    }
+
+    @Benchmark
+    fun pooleCreate() = state.forEach { (_, result) ->
+        result.pooleCreated =
+            result.pooleParsed?.let {
+                ch.poole.openinghoursparser.Util.rulesToOpeningHoursString(it)
+            }
+    }
+
+    @TearDown
+    fun teardown() {
+        println("Results:")
+        processResults()
+        println()
+        println("Poole results:")
+        processPooleResults()
+    }
+
+    private fun processResults() = printReport(
+        invalidParseable =
+            state
+                .filter { (_, result) -> result.parsed != null && !result.valid }
+                .map { (input, _) -> input }
+                .also { outputToFile(it, "invalid_but_unambiguous_opening_hours.tsv") },
+        invalidNonParseable =
+            state
+                .filter { (_, result) -> result.parsed == null }
+                .map { (input, _) -> input }
+                .also { outputToFile(it, "invalid_opening_hours.tsv") },
+        canonical =
+            state
+                .filter { (input, result) -> input.times == result.created }
+                .map { (input, _) -> input },
+    )
```

Comment:
I find it bad practice for methods to have side-effects. OK, the method is called `processResults`, but then, maybe the saving to file should happen outside the `printReport` method call. E.g. something like this (didn't test):

```suggestion
    private fun processResults() {
        val invalidParseable = state
                .filter { (_, result) -> result.parsed != null && !result.valid }
                .map { (input, _) -> input }
        val invalidNonParseable = state
                .filter { (_, result) -> result.parsed != null && !result.valid }
                .map { (input, _) -> input }
        val canonical = state.filter { (input, result) -> input.times == result.created }

        outputToFile(invalidParseable, "invalid_but_unambiguous_opening_hours.tsv")
        outputToFile(invalidNonParseable, "invalid_opening_hours.tsv")

        printReport(
            invalidParseable = invalidParseable,
            invalidNonParseable = invalidNonParseable,
            canonical = canonical,
        )
    }
```